common/src/main/java/org/mcaccess/minecraftaccess/features/point_of_interest/LockingHandler.java

```diff
@@ -126,52 +128,69 @@ private void lookAtLockedTarget() {
      */
     private void bowAimingAssist() {
         LocalPlayer player = WorldUtils.getClientPlayer();
+        if (player == null) return;
+
+        // Check if player is using a bow
         if (config.aimAssistEnabled && !aimAssistActive && player.isUsingItem() && player.getUseItem().getItem() instanceof BowItem) {
             List<Entity> hostileEntities = BuiltinEntityPOIGroups.HOSTILE.group.getItems();
-            if (!hostileEntities.isEmpty()) {
-                Entity entity = hostileEntities.stream()
-                        .min(Comparator.comparingDouble(e -> WorldUtils.getClientPlayer().distanceTo(e)))
-                        .get();
-                if (lockOnEntity(entity)) {
-                    aimAssistActive = true;
+            if (hostileEntities != null && !hostileEntities.isEmpty()) {
+                Optional<Entity> closestEntity = hostileEntities.stream()
+                        .min(Comparator.comparingDouble(player::distanceTo));
+
+                if (closestEntity.isPresent()) {
+                    Entity entity = closestEntity.get();
+                    if (lockOnEntity(entity)) {
+                        aimAssistActive = true;
+                    }
```

Comment:
`closestEntity` is always present because there is already a check that `hostileEntities` is not empty.
```suggestion
                Entity closestEntity = hostileEntities.stream()
                        .min(Comparator.comparingDouble(player::distanceTo))
                        .get();
                if (lockOnEntity(entity)) {
                    aimAssistActive = true;
```